### ISSUE TITLE
The feature "Change 'share' link to format of [post-name](url)" doesn't work anymore

### ISSUE THREAD

**Issue by phuclv90** [soscripted/sox#434]
**Describe the bug**

Pressing "Share a link to this question" shows a pure link instead of a Markdown-formatted string

**To Reproduce**
Steps to reproduce the behaviour:
1. Pressing "Share a link to this question"
2. See error

**SOX errors logged in console**

```
integrator.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
integrator.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  logging sox stored values --- 
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  access token set
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  SOX-lastVersionInstalled
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  2.3.0
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  SOXSETTINGS
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  ["Appearance-addAuthorNameToInboxNotifications","Appearance-alignBadgesByClass","Appearance-answerTagsSearch","Appearance-colorAnswerer","Appearance-dragBounty","Appearance-highlightQuestions","Appearance-isQuestionHot","Appearance-localTimestamps","Appearance-markEmployees","Appearance-metaChatBlogStackExchangeButton","Appearance-metaNewQuestionAlert","Appearance-spoilerTip","Appearance-standOutDupeCloseMigrated","Appearance-tabularReviewerStats","Appearance-topAnswers","Appearance-unspoil","Appearance-addTimelineAndRevisionLinks","Appearance-showTagWikiLinkOnTagPopup","Comments-autoShowCommentImages","Comments-commentReplies","Comments-commentShortcuts","Comments-confirmNavigateAway","Comments-copyCommentsLink","Comments-moveBounty","Comments-showCommentScores","Comments-hiddenCommentsIndicator","Comments-onlyShowCommentActionsOnHover","Editing-addSBSBtn","Editing-editComment","Editing-editReasonTooltip","Editing-findAndReplace","Editing-kbdAndBullets","Editing-titleEditDiff","Editing-inlineEditorEverywhere","Editing-pasteImagesDirectly","Flags-flagOutcomeTime","Flags-flagPercentages","Flags-flagPercentageBar","Sidebar-hideCommunityBulletin","Sidebar-hideChatSidebar","Sidebar-linkedToFrom","Sidebar-hotNetworkQuestionsFiltering","Chat-chatEasyAccess","Chat-replyToOwnChatMessages","Chat-renameChat","Voting-betterCSS","Voting-stickyVoteButtons","Voting-disableVoteButtons","Extras-alwaysShowImageUploadLinkBox","Extras-linkedPostsInline","Extras-parseCrossSiteLinks","Extras-quickAuthorInfo","Extras-shareLinksMarkdown","Extras-sortByBountyAmount","Extras-warnNotLoggedIn","Extras-showMetaReviewCount","Extras-copyCode","Extras-dailyReviewBar","Extras-openLinksInNewTab","Extras-showQuestionStateInSuggestedEditReviewQueue"]
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  editReasons
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  [["formatting","Improved Formatting"],["spelling","Corrected Spelling"],["grammar","Fixed grammar"],["greetings","Removed thanks/greetings"]]
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  metaNewQuestionAlert-lastQuestions
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  {"meta.security":"Would this question be on-topic here?","meta.superuser":"Drop kick [dropout]","meta.stackoverflow":"How can I fix my bad question?"}
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  SOX-debug
userscript.html?id=1614e980-0234-4a57-9e93-c9aa4fd93b82:68 SOX:  true
securepubads.g.doubleclick.net/gampad/ads?gdfp_req=1&pvsid=1824489043743300&correlator=4487848526350978&output=ldjh&callback=googletag.impl.pubads.callbackProxy1&impl=fifs&json_a=1&eid=21064439%2C21064521&vrg=2019082701&guci=2.2.0.0.2.2.0.0&plat=1%3A32776%2C2%3A16809992%2C8%3A134250504&sc=1&sfv=1-0-35&ecs=20190903&iu_parts=248424177%2Csuperuser.com%2Csb%2Cquestion-pages&enc_prev_ius=%2F0%2F1%2F2%2F3&prev_iu_szs=300x250%7C300x600&cust_params=Registered%3Dtrue%26Reputation%3D10k%26ron-tag%3Dmicrosoft-excel%252Cmicrosoft-excel-2016%26IsHighRepEarner%3Dtrue%26Sidebar%3DRight&cookie=ID%3D8cfe03c588074f41%3AT%3D1558521797%3AS%3DALNI_Mbsj--i9EWijzN6aRG_P0m8jW-tqQ&bc=31&abxe=1&lmt=1567476372&dt=1567476372395&dlt=1567476371825&idt=562&frm=20&biw=1903&bih=937&oid=3&adxs=-12245933&adys=-12245933&adks=2001992165&ucis=1&ifi=1&u_tz=420&u_his=1&u_h=1080&u_w=1920&u_ah=1040&u_aw=1920&u_cd=24&u_nplug=3&u_nmime=4&u_sd=1&flash=0&url=https%3A%2F%2Fsuperuser.com%2Fquestions%2F1478233%2Fgiven-n-number-of-cells-is-there-a-way-to-fill-in-n-1-cells-and-the-nth-cell-w&ref=https%3A%2F%2Fsuperuser.com%2Fquestions%2F1478233%2Fgiven-n-number-of-cells-is-there-a-way-to-fill-in-n-1-cells-and-the-nth-cell-w&dssz=27&icsg=2690&std=0&vis=1&dmc=8&scr_x=0&scr_y=159&psz=300x938&msz=0x0&blev=1&bisch=1&ga_vid=1081730392.1558521798&ga_sid=1567476372&ga_hid=1244665167&fws=128&ohw=0:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
tpc.googlesyndication.com/safeframe/1-0-35/html/container.html:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
```

**Expected behaviour**

The share link is formatted properly

**Screenshots/GIFs**
If applicable, please add screenshots/GIFs (you can use something like [LICEcap](https://www.cockos.com/licecap/) to make one!) to help explain the problem.

**Environment**

* Chrome 76.0.3809.132 64-bit, Tampermonkey 4.8.41
* Latest Firefox (not on it right now so version is unclear)

---

### Features Enabled 

    ["Appearance-addAuthorNameToInboxNotifications","Appearance-alignBadgesByClass","Appearance-answerTagsSearch","Appearance-colorAnswerer","Appearance-dragBounty","Appearance-highlightQuestions","Appearance-isQuestionHot","Appearance-localTimestamps","Appearance-markEmployees","Appearance-metaChatBlogStackExchangeButton","Appearance-metaNewQuestionAlert","Appearance-spoilerTip","Appearance-standOutDupeCloseMigrated","Appearance-tabularReviewerStats","Appearance-topAnswers","Appearance-unspoil","Appearance-addTimelineAndRevisionLinks","Appearance-showTagWikiLinkOnTagPopup","Comments-autoShowCommentImages","Comments-commentReplies","Comments-commentShortcuts","Comments-confirmNavigateAway","Comments-copyCommentsLink","Comments-moveBounty","Comments-showCommentScores","Comments-hiddenCommentsIndicator","Comments-onlyShowCommentActionsOnHover","Editing-addSBSBtn","Editing-editComment","Editing-editReasonTooltip","Editing-findAndReplace","Editing-kbdAndBullets","Editing-titleEditDiff","Editing-inlineEditorEverywhere","Editing-pasteImagesDirectly","Flags-flagOutcomeTime","Flags-flagPercentages","Flags-flagPercentageBar","Sidebar-hideCommunityBulletin","Sidebar-hideChatSidebar","Sidebar-linkedToFrom","Sidebar-hotNetworkQuestionsFiltering","Chat-chatEasyAccess","Chat-replyToOwnChatMessages","Chat-renameChat","Voting-betterCSS","Voting-stickyVoteButtons","Voting-disableVoteButtons","Extras-alwaysShowImageUploadLinkBox","Extras-linkedPostsInline","Extras-parseCrossSiteLinks","Extras-quickAuthorInfo","Extras-shareLinksMarkdown","Extras-sortByBountyAmount","Extras-warnNotLoggedIn","Extras-showMetaReviewCount","Extras-copyCode","Extras-dailyReviewBar","Extras-openLinksInNewTab","Extras-showQuestionStateInSuggestedEditReviewQueue"]

**Comment by double-beep** [comment]
You appear to use SOX version 2.3.0 which is a bit old. Please try [the latest DEV version](https://github.com/soscripted/sox/raw/dev/sox.user.js) and tell me if it happens again. It should have been fixed by #431

**Comment by phuclv90** [comment]
I've just tried V2.5.31DEV and somehow it works in some communities like superuser or software recommendations but not in stackoverflow or askubuntu

**Comment by double-beep** [comment]
SOX errors in console? I cannot reproduce the issue on Stack Overflow or Ask Ubuntu with the options you use. Give also specific links to questions it doesn't work! If nothing works, then make sure to disable all userscripts except SOX and check for conflicts (i.e. does SOX work when this userscript is disabled?)

**Comment by phuclv90** [comment]
I tried refreshing the page and restarting the browser and found out that it doesn't occur only on stackoverflow and askubuntu but any SE sites but only on some questions. Some examples where the issue occurs:

* https://stackoverflow.com/questions/176264/what-is-the-difference-between-a-uri-a-url-and-a-urn
* https://stackoverflow.com/questions/1642028/what-is-the-operator-in-c
* https://askubuntu.com/questions/1170334/sound-do-not-work-and-sound-test-fail-hp-notebook
* https://superuser.com/q/1477967/241386

Logs for them:

* [stackoverflow.com-1567491026486.log](https://github.com/soscripted/sox/files/3567984/stackoverflow.com-1567491026486.log)
* [stackoverflow.com-1567491198929.log](https://github.com/soscripted/sox/files/3567992/stackoverflow.com-1567491198929.log)
* [askubuntu.com-1567491491617.log](https://github.com/soscripted/sox/files/3568015/askubuntu.com-1567491491617.log)
* [superuser.com-1567491657521.log](https://github.com/soscripted/sox/files/3568014/superuser.com-1567491657521.log)

I've also realized that if I close a question where the feature works then reopen the tab with Ctrl+Shift+T then the link will never be replaced anymore. For example https://unix.stackexchange.com/questions/538774/how-do-i-convert-into-a-regular-expression with the log [unix.stackexchange.com-1567491860179.log](https://github.com/soscripted/sox/files/3568021/unix.stackexchange.com-1567491860179.log)

**Comment by shu8** [comment]
Thanks for investigating. Weirdly, I could reproduce this on all questions I tried, not just the ones you posted. It seems like it's a timing issue, which should be fixed in [dev 2.5.32](https://github.com/soscripted/sox/raw/dev/sox.user.js) now :)

Please could you confirm it's working for you?

**Comment by phuclv90** [comment]
Looks like it's working now
